src/Node/ReadLine.purs

```diff
@@ -1,42 +1,88 @@
 -- | This module provides a binding to the Node `readline` API.
 
-module Node.ReadLine where
+module Node.ReadLine
+  ( Interface
+  , InterfaceOptions
+  , Completer
+  , LineHandler
+  , createInterface
+  , createConsoleInterface
+  , noCompletion
+  , prompt
+  , setPrompt
+  , setLineHandler
+  , close
+  ) where
 
 import Prelude (return)
-
-import Control.Monad.Eff
-import Control.Monad.Eff.Console
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Node.Process (stdin, stdout)
+import Node.Stream (Readable, Writable)
 
 -- | A handle to a console interface.
 -- |
 -- | A handle can be created with the `createInterface` function.
 foreign import data Interface :: *
 
+foreign import createInterfaceImpl :: forall opts eff. {| opts} -> Eff eff Interface
+
+foreign import undefined :: forall a. a
+
+-- | Options passed to `readline`'s `createInterface`
+type InterfaceOptions r w e =
+  { input :: Readable r e
+  , output :: Maybe (Writable w e)
+  , completer :: Maybe (Completer e)
+  , terminal :: Maybe Boolean
+  , historySize :: Maybe Int
+  }
+
 -- | A function which performs tab completion.
 -- |
--- | This function takes the partial command as input, and returns a collection of 
+-- | This function takes the partial command as input, and returns a collection of
 -- | completions, as well as the matched portion of the input string.
-type Completer eff = String -> Eff (console :: CONSOLE | eff) { completions :: Array String, matched :: String }
+type Completer eff = String -> Eff eff { completions :: Array String, matched :: String }
 
--- | A function which handles input from the user.
-type LineHandler eff a = String -> Eff (console :: CONSOLE | eff) a 
+-- | Builds an interface with the specified options.
+createInterface :: forall r w eff. InterfaceOptions r w eff -> Eff eff Interface
+createInterface opts = createInterfaceImpl
+  { input: opts.input
+  , output: fromMaybe undefined opts.output
+  , completer: fromMaybe undefined opts.completer
+  , terminal: fromMaybe undefined opts.terminal
+  , historySize: fromMaybe undefined opts.historySize
+  }
 
--- | Set the current line handler function.
-foreign import setLineHandler :: forall eff a. Interface -> LineHandler eff a -> Eff (console :: CONSOLE | eff) Interface
+-- | Create an interface with the specified completion function.
+createConsoleInterface :: forall eff.
+                          Completer (console :: CONSOLE, err :: EXCEPTION | eff)
+                       -> Eff (console :: CONSOLE, err :: EXCEPTION | eff) Interface
```

Comment:
`stdout` is a `forall eff. Writable () (console :: CONSOLE, err :: EXCEPTION | eff)`